### PR TITLE
cli: enable cleaning of parts

### PR DESCRIPTION
**Pull request by sergiusens**
Bring back snapcraft clean <part-name> when using build providers.

LP: #1779136
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [ ] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Comment by kenvandine** [comment]
Looking pretty good, however, I ran into an issue with cleaning parts with dependent parts.  

  File "/snap/snapcraft/x1/lib/python3.5/site-packages/snapcraft/internal/project_loader/_parts_config.py", line 167, in clean_part
    part.clean(staged_state, primed_state, step)
  File "/snap/snapcraft/x1/lib/python3.5/site-packages/snapcraft/internal/pluginhandler/__init__.py", line 898, in clean
    self._clean_steps(project_staged_state, project_primed_state, step)
  File "/snap/snapcraft/x1/lib/python3.5/site-packages/snapcraft/internal/pluginhandler/__init__.py", line 925, in _clean_steps
    self.clean_prime(project_primed_state)
  File "/snap/snapcraft/x1/lib/python3.5/site-packages/snapcraft/internal/pluginhandler/__init__.py", line 831, in clean_prime
    self._clean_shared_area(self.primedir, state, project_primed_state)
  File "/snap/snapcraft/x1/lib/python3.5/site-packages/snapcraft/internal/pluginhandler/__init__.py", line 852, in _clean_shared_area
    _clean_migrated_files(primed_files, primed_directories, shared_directory)
  File "/snap/snapcraft/x1/lib/python3.5/site-packages/snapcraft/internal/pluginhandler/__init__.py", line 1133, in _clean_migrated_files
    os.remove(os.path.join(directory, snap_file))
FileNotFoundError: [Errno 2] No such file or directory: '/root/prime/usr/share/icons/Humanity/mimes/16/libreoffice-oasis-web-template.png'

File: snapcraft/cli/lifecycle.py [hunk 1]
```diff
@@ -273,20 +273,13 @@ def pack(directory, output, **kwargs):
 
 @lifecyclecli.command()
 @click.argument("parts", nargs=-1, metavar="<part>...", required=False)
-@click.option(
-    "--step",
```
Comment (kyrofa) [inline snapcraft/cli/lifecycle.py]: The ability to clean specific steps is SUPER helpful. For example, it's making debugging this PR hard :) .

Comment (sergiusens) [reply snapcraft/cli/lifecycle.py]: yes, read the bug description

**Comment by cmatsuoka** [comment]
Something I missed when building packages was the ability to run a full clean inside the multipass VM instead of nuking the entire VM (slow to recreate), so maybe we could have two "levels" of clean: one that recreates the build provider and other that acts inside the existing build provider. This commit helps a lot, but still requires parts to be specified.

**Pull request merged**